peps/pep-9999.rst

```diff
@@ -0,0 +1,300 @@
+``Project-URL`` equivalents.
+
+This formalization has two parts:
+
+1. A set of rules for canonicalizing ``Project-URL`` labels;
+2. A set of "well-known" canonical label values that indices may specialize
+   URL presentation for.
+
+Label canonicalization
+----------------------
+
+The core metadata specification stipulates that ``Project-URL`` labels are
+free text, limited to 32 characters.
+
+This PEP proposes adding the concept of a "canonicalized" label to the core
+metadata specification. Label canonicalization is defined via the following
+Python function:
+
+.. code-block:: python
+
+    def canonicalize_label(label: str) -> str:
+        return label.lower().replace("-", "").replace("_", "").replace(" ", "")
+
+In plain language: a label is *canonicalized* by deleting all spans of dashes,
+underscores, and spaces.
+
+The following table shows examples of labels before (raw) and after
+canonicalization:
+
+.. csv-table::
+    :header: "Raw", "Canonicalized"
+
+    "``Homepage``", "``homepage``"
+    "``Home-page``", "``homepage``"
+    "``Home page``", "``homepage``"
+    "``Change_Log``", "``changelog``"
+
+Metadata producers **SHOULD** emit the canonicalized form of a user
+specified label, but **MAY** choose to emit the un-canonicalized form so
+long as it adheres to the existing 32 character constraint.
+
+Well-known labels
+-----------------
+
+In addition to the canonicalization rules above, this PEP proposes a
+fixed (but extensible) set of "well-known" ``Project-URL`` labels,
+as well as equivalent aliases.
+
+The following table lists these labels, in canonical form:
+
+.. csv-table::
+    :header: "Label", "Description", "Aliases"
+    :widths: 20, 60, 20
+
+    "``homepage``", "The project's homepage", "*(none)*"
+    "``download``", "A download URL for the current distribution, equivalent to ``Download-URL``", "*(none)*"
+    "``changelog``", "The project's changelog", "``changes``, ``releasenotes``, ``whatsnew``, ``history``"
+    "``documentation``", "The project's online documentation", "``docs``"
+    "``bug``", "The project's bug tracker", "``issue``, ``tracker``, ``report``"
+    "``funding``", "Sponsoring information", "``sponsor``, ``donate``, ``donation``"
+
+Packagers and metadata producers **MAY** choose to use these well-known
+labels to communicate specific URL intents to package indices and downstreams.
+
+Similarly, indices **MAY** choose to specialize their rendering or presentation
+of URLs with these labels, e.g. by presenting an appropriate icon or tooltip
+for each label.
+
+This PEP recognizes that the list of well-known labels is unlikely to remain
+static, and that subsequent additions to it should not require the overhead
+associated with a formal PEP process or new metadata version. Consequently,
+this PEP proposes that the list above become a "living" list within the
+living core metadata standard, which PyPA members may amend via ordinary
+contributing processes.
```

Comment:
Rather than doing that, perhaps we should delegate the role of documenting the living conventions specifically to the Python Package Index documentation?

```suggestion
associated with a formal PEP process or new metadata version. As the primary
expected use case for this information is to control the way project URLs are
displayed on the Python Package Index, this PEP proposes that the list above
become a "living" list within PyPI's documentation (at time of writing, the
documentation for influencing PyPI's URL display can be found
`here <https://docs.pypi.org/project_metadata/#icons>`__).
```